corehq/ex-submodules/pillowtop/processors/elastic.py

```diff
@@ -159,30 +162,74 @@ def process_changes_chunk(self, changes_chunk):
         return retry_changes, error_changes
 
 
-def send_to_elasticsearch(alias, doc_type, doc_id, es_getter, name, data=None,
-                          retries=MAX_RETRIES, propagate_failure=settings.UNIT_TESTING,
-                          update=False, delete=False, es_merge_update=False):
+@quickcache(['alias'], memoize_timeout=24 * 60 * 60, timeout=24 * 60 * 60)
+def _get_indices_by_alias(alias):
+    # The cache is not invalidated. So be cautious when using this.
+    #   This designed to be used with ILM indices, where index names
+    #   are of a predictable format such as xforms_2016-07-07-000002, xforms_2016-07-07-000001 etc
+    from corehq.elastic import get_es_new
+    es = get_es_new()
+    return list(es.indices.get_alias(alias))
+
+
+def _doc_exists_in_es(doc_id, doc_type, index_info, es_interface):
+    # for ILM indices returns a tuple of (whether doc exists, the index that the doc is found in)
+    # for normal indices returns a tuple of (whether doc exists, None)
+    if not index_info.ilm_config:
+        return es_interface.doc_exists(index_info.alias, doc_id, doc_type), None
+    # exists/get queries doesn't work against aliases
+    #   so query all backing indices
+    if settings.UNIT_TESTING:
+        _get_indices_by_alias.clear(index_info.alias)
+    indices = _get_indices_by_alias(index_info.alias)
+    # todo; _get_indices_by_alias could be stale
+    #   so we could query the next index based on pattern even if it doesn't exist
+    doc_exists = False
+    found_in = None
+    for index in indices:
```

Comment:
wouldn't it be more efficient to do a search query to find the index? Also if the doc does exists we will need to get the `seq_no` and `primary_term` in order to do the update https://www.elastic.co/guide/en/elasticsearch/reference/current/use-a-data-stream.html#update-delete-docs-in-a-backing-index

Reply:
> wouldn't it be more efficient to do a search query to find the index? 
The search query on the alias pointing to all indices is more expensive than a get query on individual indices.

>  Also if the doc does exists we will need to get the seq_no and primary_term in order to do the update

We are not using a data stream, so this doesn't apply in this case.